qa/mc_sc_connected_nodes.py

```diff
@@ -80,63 +80,69 @@ def run_test(self):
-                                       [genesis_account[1]], [wallet_balance], False)
-
-        try:
-            check_mainchan_block_inclusion(third_sidechain_node, sidechain_id, 2, 0, first_mainchain_node_new_block,
-                                           [genesis_account[1]], [wallet_balance], False)
-            # SC node 2 should not to know information about blocks generated from MC node 0
-            # Otherwise the test must to fail
-            assert_equal(1, 2)
-        except AssertionError:
-            print "SC node 2 doesn't include block generated from MC node 0."
-            connect_nodes_bi(self.nodes, 0, 1)
-            self.sync_all()
-            connect_sc_nodes(self.sc_nodes[0], 1)
-            connect_sc_nodes(self.sc_nodes[0], 2)
-            self.sc_sync_all()
-            check_mainchan_block_inclusion(third_sidechain_node, sidechain_id, 2, 0, first_mainchain_node_new_block,
-                                            [genesis_account[1]], [wallet_balance], False)
-            block_hash = second_mainchain_node.generate(1)
-            self.sync_all()
-            second_mainchain_node_new_block = second_mainchain_node.getblock(block_hash[0])
-            third_sidechain_node.block_generate(json.dumps({"number":1}))
-            self.sc_sync_all()
-            check_mainchan_block_inclusion(first_sidechain_node, sidechain_id, 3, 0, second_mainchain_node_new_block,
-                                           [genesis_account[1]], [wallet_balance], False)
-            check_mainchan_block_inclusion(second_sidechain_node, sidechain_id, 3, 0, second_mainchain_node_new_block,
-                                           [genesis_account[1]], [wallet_balance], False)
-            check_mainchan_block_inclusion(third_sidechain_node, sidechain_id, 3, 0, second_mainchain_node_new_block,
-                                           [genesis_account[1]], [wallet_balance], False)
 
+        # SC node 1 and 2 forge 1 SC block
+        sc_generate_blocks(first_sidechain_node)
+        sc_generate_blocks(second_sidechain_node)
+
+        # verify the block is included inside SC nodes 1 and 2
+        sc_1_mc_block_inclusion = is_mainchain_block_included(first_sidechain_node, sidechain_id, 2, 0, first_mainchain_node_new_block)
+        sc_2_mc_block_inclusion = is_mainchain_block_included(second_sidechain_node, sidechain_id, 2, 0, first_mainchain_node_new_block)
+        assert_true(sc_1_mc_block_inclusion, "The mainchain block is not included for SC node 1.")
+        assert_true(sc_2_mc_block_inclusion, "The mainchain block is not included for SC node 2.")
+
+        # verify the mc block is NOT included inside SC node 3
+        sc_3_mc_block_inclusion = is_mainchain_block_included(third_sidechain_node, sidechain_id, 2, 0, first_mainchain_node_new_block)
+        assert_false(sc_3_mc_block_inclusion, "The mainchain block is included for SC node 3.")
+
+        # connect MC 1 to MC 2
```

Comment:
please, review the test workflow. Seems, that next actions flow is enough:
1. 2 mc nodes isolated, 2 sc nodes isolated and connected to different mc nodes.
2. verify genesis blocks
3. first mc node generates a block
4. second mc node should be unchanged
5. first sc node generates a block and verify it against first mc node new block.
6. second sc node should be unchanged
7. connect mc nodes and synchronize and verify changes in second mc node
8. second sc node generates a block and verify it against second mc node best block.
9. first sc node generates a block: verify that second sc is not changed
10. connect sc nodes, sync and see that bost sc nodes has best block from previous point.

May be even more minimalistic. At least third sc node doesn't give any benefit.

Reply:
Why?